### PR TITLE
Selectively skip and fix tests after PR#212

### PR DESCRIPTION
**Pull request by dliappis**
PR #212 made some changes on `main` that require the presence of `MANIFEST_URL`. However, the pipeline is still configured to run the same script when a PR (from an upstream branch) is raised and it currently fails.

This commit skips tests when triggered from another pipeline (unified release), and allows running tests when triggered by PR creation, or branch pushed by picking up a suitable `MANIFEST_URL`.

### CODE REVIEW

File: .buildkite/scripts/build.ps1 [hunk 1]
```diff
@@ -1,7 +1,35 @@
 $ErrorActionPreference = "Stop"
 Set-Strictmode -version 3
 
-if (-not (Test-Path env:MANIFEST_URL)) {
+if (-not (Test-Path env:MANIFEST_URL) -and (Test-Path env:BUILDKITE_PULL_REQUEST) -and ($env:BUILDKITE_PULL_REQUEST -ne "false")) {
+    # we are called via a PR
+    Write-Host "~~~ Running in pull request mode"
+
+    $targetBranch = $env:BUILDKITE_PULL_REQUEST_BASE_BRANCH
+    if ( ($targetBranch -ne "main") -and -not ($targetBranch -like "8*")) {
```
Comment (amitkanfer) [inline .buildkite/scripts/build.ps1:9]: perhaps a nit, but should we extract the `8` to a constant at the top of the file?

Comment (dliappis) [reply .buildkite/scripts/build.ps1:9]: Good idea, will do.

**Comment by dliappis** [comment]
This is ready for review. Because of the two triggers that are currently configured for this job (`branch commits` and `pr commits`) we need to support a different way of running depending on the occasion.

So if we push e.g. a branch like `8.13` it will run the tests as shown in https://buildkite.com/elastic/elastic-stack-installers/builds/3647#018d5b6b-d239-4547-badd-c351d2a18308:

image

If we open the PR it (e.g. the latest commit here: https://buildkite.com/elastic/elastic-stack-installers/builds/3646) it looks like:

image

If the pipeline gets triggered by the unified release it should run as before (and allow to go through the DRA publish).

**Comment by dliappis** [comment]
@strawgate to implement what we discussed in do you think it would it be enough to just present a variable [here](https://github.com/elastic/elastic-stack-installers/blob/43efe9d1058f0e8b82da0876b2f5ba2784834710/.buildkite/scripts/build.ps1#L62) (which is the conditional that covers the case that our job/scripts gets triggered from the unified release) and conditionally run tests in https://github.com/elastic/elastic-stack-installers/blob/d1bdb21c8a0e159e079bda52186e5bed8b542187/.buildkite/scripts/build.ps1#L185 only if the var is present?

**Comment by strawgate** [comment]
> @strawgate to implement what we discussed in do you think it would it be enough to just present a variable [here](https://github.com/elastic/elastic-stack-installers/blob/43efe9d1058f0e8b82da0876b2f5ba2784834710/.buildkite/scripts/build.ps1#L62) (which is the conditional that covers the case that our job/scripts gets triggered from the unified release) and conditionally run tests in
> 
> https://github.com/elastic/elastic-stack-installers/blob/d1bdb21c8a0e159e079bda52186e5bed8b542187/.buildkite/scripts/build.ps1#L185
> 
> only if the var is present?

Yes!

**Comment by dliappis** [comment]
> > @strawgate to implement what we discussed in do you think it would it be enough to just present a variable [here](https://github.com/elastic/elastic-stack-installers/blob/43efe9d1058f0e8b82da0876b2f5ba2784834710/.buildkite/scripts/build.ps1#L62) (which is the conditional that covers the case that our job/scripts gets triggered from the unified release) and conditionally run tests in
> > https://github.com/elastic/elastic-stack-installers/blob/d1bdb21c8a0e159e079bda52186e5bed8b542187/.buildkite/scripts/build.ps1#L185
> > 
> > only if the var is present?
> 
> Yes!

@strawgate this has been implemented in b1c05bae4746870179483a0f75a16e8d54569a02 . Please take a look.

File: .buildkite/scripts/build.ps1 [hunk 2]
```diff
@@ -1,11 +1,68 @@
...
+
+function setManifestUrl {
+    param (
+        [string]$targetBranch
+    )
+
+    # the snapshot API url expects master where we normally use main
+    $snapshotsApiTargetBranch = if ($targetBranch -eq "main") { "master" } else { $targetBranch }
+
+    $snapshotsUrl = "https://snapshots.elastic.co/latest/$snapshotsApiTargetBranch.json"
```
Comment (dliappis) [inline .buildkite/scripts/build.ps1:15]: Remember to change to staging based on `DRA_WORKFLOW`

**Comment by dliappis** [comment]
close https://github.com/elastic/elastic-stack-installers/pull/225 in favor of https://github.com/elastic/elastic-stack-installers/pull/235

**Pull request closed without merging**